PROTO_tests/tests/groupby_test.py

```diff
@@ -654,6 +687,82 @@ def test_large_mean_aggregation(self):
         for m in means.to_list():
             assert np.isclose(float(a[0]), m)
 
+    #   ak.unique takes 1 pda argument and 3 booleans
+    #      However, not all 8 combinations of the booleans are needed to
+    #      cover the test space.
+    #      Combinations TTF and TFF are supersets of all other possible
+    #      combinations, so only those are tested below.
+
+    @pytest.mark.parametrize("data_type", UNIQUE_TYPES)
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_unique(self, data_type, prob_size):
+        Jenny = pytest.seed if pytest.seed is not None else 8675309
+        T = True
+        F = False
+        np.random.seed(Jenny)
+        arrays = {
+            npstr: np.random.choice(VOWELS_AND_SUCH, prob_size),
+            ak.int64: np.random.randint(0, prob_size // 3, prob_size),
+            ak.float64: np.random.uniform(0, prob_size // 3, prob_size),
+            ak.categorical: np.random.choice(PICKS, prob_size),
+        }
+        nda = arrays[data_type]
+        np_unique = np.unique(nda)  # get unique keys from np for comparison
+        s_nda, us_nda = make_sorted_and_unsorted_data(nda)
+        s_pda = ak.array(s_nda)
+        us_pda = ak.array(us_nda)
+
+        # Categorical requires another step to make the pdarrays categorical
+
+        if data_type == "categorical":
+            s_pda = ak.Categorical(s_pda)
+            us_pda = ak.Categorical(us_pda)
+
+        # Call ak.unique with TTF and TFF
+
+        ak_TTF = ak.unique(s_pda, T, T, F)
+        ak_TFF = ak.unique(us_pda, T, F, F)
+
+        # Check for correct unique keys.
+
+        assert np.all(np_unique == np.sort(ak_TFF[0].to_ndarray())) and np.all(
+            np_unique == np.sort(ak_TTF[0].to_ndarray())
+        )
+
+        # Check groups and indices.  If data was sorted, the group ndarray
+        # should just be list(range(len(nda))), and the indices will be
+        # exactly what the list method returns as the first index
+
+        srange = np.array(list(range(len(nda))))
+        check1 = np.all(srange == ak_TTF[1].to_ndarray())
+        indices = ak_TTF[2]
+        check2 = T
+        nda_list = list(s_nda)
+        for i in range(len(indices)):
+            check2 = check2 and indices[i] == nda_list.index(np_unique[i])
+        assert check1 and check2
+
+        # for unsorted data, a bit more work is required.  A reordered
+        # copy of the pdarray is created based on the returned permutation,
+        # and the indices are used to confirm that elements within a segment
+        # match.
+
+        aku = ak.unique(us_pda).to_ndarray()
+        reordering = ak_TFF[1].to_ndarray()
+        reordered = [nda[i] for i in reordering]
```

Comment:
If you don't convert to ndarray first, this becomes: 

```
aku = ak.unique(us_pda)
reordered = us_pda[ak_TFF[1]]
ak.all (reordered == ak.broadcast(indices, aku, ak_TFF[1].size) )
```

So 15 lines can be done in 3 lines....

Reply:
You might also be able to simplify the one above with something similar:
`ak.broadcast(indices, ak.array(np_unique), len(nda_list))`

Reply:
In your first comment above, do you mean "assert ak.all (...)"

Reply:
Writing it that way (assert ak.all(...) ) passes on categoricals, ints and floats, but fails on strings.  I'm going to revert for now.

Reply:
Okay, my mistake.  I had one lousy typo.  Your suggestions both work, and make the code much more concise.